### PR TITLE
Dev/design and implementation

### PR DESCRIPTION
**Pull request by Bluntsord**

### CODE REVIEW

**Review by bharathcs — CHANGES_REQUESTED** [review]
* Is the `getLogicAction()` overlooked for DeleteSequenceDiagram on purpose? Its fine to skip on DeleteSequenceDiagramUiPart, its cleaner that way , but I do believe you should check out how I made the SwitchTabCommand sequence diagram
* ParserUtil has a dependency to ArgumentMultimap
* Thanks for trimming whitespace hehe
* Do merge master :)

**Review by bharathcs — APPROVED** [review]
Nice!!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -248,10 +237,157 @@ Here is an example of a `Person` in JSON form:
...
+
+<div markdown="span" class="alert alert-warning">**Explanation of ParserContext:**
+<br>
+1. When a `XYZCommand` class (e.g. `HomeCommand`, `ViewCommand`,...) is executed, it returns a `CommandResult` object containing a logic action if the `XYZCommand` requires a change in tab or view. <br>
+2. `LogicManager` accepts this `CommandResult` object and executes the logic action here.`LogicManager` is a facade that is able to set and change the current `ParserStrategy`.<br>
+3. `ParserContext` in `LogicManager` is updated to contain the `ABCParser` of the new view or tab.<br>
+4. `UI` is updated to change its state, which is kept track of by `UiState` by accepting the consumer also in the command result.<br>
+</div>
+
+#### 4.3 NotesTab
```
Comment (sheehui) [inline docs/DeveloperGuide.md:379]: camel case

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -248,10 +237,157 @@ Here is an example of a `Person` in JSON form:
...
+
+The notes tab is different from the commands above. Instead of being command-based, the notes tab allows the user
```
Comment (sheehui) [inline docs/DeveloperGuide.md:381]: a tab being different from the command? might want to rephrase this para

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -248,10 +237,157 @@ Here is an example of a `Person` in JSON form:
...
+to type in any quick notes that the user would want. Updating it in realtime. This is opposed to the commands where,
+changes to any other component only happen when the command is executed. Hence the notes tab take advantage of a
```
Comment (sheehui) [inline docs/DeveloperGuide.md:383]: takes advantage of

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -248,10 +237,157 @@ Here is an example of a `Person` in JSON form:
...
+3. Remove<br>
+</div>
+
+Edit command is a command that edits the information of a specific client. Other commands like append and remove,
+also deal directly with a specific client's information. Thus they fall under the second [category](#42-implementation-and-commands)
+Like other commands in the other 2 [categories](#42-implementation-and-commands), edit command follows the same general
+structure. We will be using the `Edit` command as the example to illustrate and explain all commands under this category.
+
+Explanation:
+* The `UI` takes in the command inputted from the user and passes it to the `Logic` component.
```
Comment (sheehui) [inline docs/DeveloperGuide.md:322]: takes in the input command from

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -248,10 +237,157 @@ Here is an example of a `Person` in JSON form:
...
+* The `Logic` component parses the command and returns the `Delete` command.
+* The `Logic` component executes the `Delete` command. The `deletePerson` method in `Model` is called which then deletes the `Person` object p from `donnafin.json`.
+* The `Logic` component then accepts the LogicConsumer produced from the command result. This consumer will alter the logic component depending on the command result. In this case, for the `delete` command, the consumer makes no change to logic.
+* The `Logic` component then continues with the `execute` command and calls the `saveAddressBook` method to save the updated `addressBook` with the deleted person.
+This does not involve a consumer in any way and is always part of execute command.
+* The `Model` component then calls `saveAddressBook` method that engages the `Storage` component to save the updated changes to storage locally.
+* The `UI` component then accepts the UiConsumer produced from the command result. This consumer will alter the UI component depending on the command result. In this case, for the `delete` command, the consumer makes no change to logic.
+
+#### 4.2.2 Commands that accesses one specific client's information
```
Comment (sheehui) [inline docs/DeveloperGuide.md:298]: commands that access

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -248,10 +237,157 @@ Here is an example of a `Person` in JSON form:
...
+1. Commands that involve organisation of clients.
+2. Commands that involve editing of personal information of one specific client,
+be it appending or removing information.
+3. Commands that involve changing of tab
+
+Despite falling under the three broad categories, the commands still have many similarities. Thus, we give an
+in depth explanation of how the first category works. Subsequent explanation of commands from the other 2 categories will follow
+the same framework but differ slightly.
+
+#### 4.2.1 Commands that involve organisation of clients
```
Comment (sheehui) [inline docs/DeveloperGuide.md:261]: Commands that involve the organisation of clients

**Review by sheehui — COMMENTED** [review]
some minor changes pls do so in a new PR and also rmb capitalise the section name in your "this is the end of xx ? click here to..." in the UG

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Bluntsord) [reply docs/DeveloperGuide.md:381]: Rephrased. The notes tab accepts user input like the commands but is fundamentally different from it.

**Pull request merged**